tools/recolor_season_variants.py

```diff
@@ -1,77 +1,124 @@
-import glob, os, shutil #pip install glob3
-import numpy as np #pip install numpy
-from PIL import Image
+"""
+Create recolored seasonal variations for sprites in all subfolders
+"""
 import sys
+import os
+import shutil
+
+import glob  # pip install glob3
+import numpy  # pip install numpy  # TODO: versions
+
+from PIL import Image
+
 
-numberPrefixes = ["_0", "_1", "_2", "_3", "_4", "_5", "_6", "_7", "_8"]
-seasonPrefixes = {
-  "_season_autumn": [(230,227,171), (216,147,110), (163,100,92), (90,68,72)],
-  "_season_winter": [(254,254,254), (203,200,218), (137,119,142), (90,68,72)],
-  "_season_summer": [(255,216,148), (194,176,107), (83,138,106), (73,75,88)]
+NUMBER_PREFIXES = ['_0', '_1', '_2', '_3', '_4', '_5', '_6', '_7', '_8']
+SEASON_PREFIXES = {
+    '_season_autumn': [
+        (230, 227, 171),
+        (216, 147, 110),
+        (163, 100, 92),
+        (90, 68, 72),
+    ],
+    '_season_winter': [
+        (254, 254, 254),
+        (203, 200, 218),
+        (137, 119, 142),
+        (90, 68, 72),
+    ],
+    '_season_summer': [
+        (255, 216, 148),
+        (194, 176, 107),
+        (83, 138, 106),
+        (73, 75, 88),
+    ],
 }
-defaultColors = [(240,236,187), (181,175,105), (115,130,92), (95,76,53)]
-
-def replaceColors(seasonFile, season):
-    replacementColors = seasonPrefixes[season]
-    img = Image.open(seasonFile).convert('RGB')
-    data = np.array(img)
-    currentColor = 0
-    
-    for color in defaultColors:
-        data[(data == color).all(axis = -1)] = replacementColors[currentColor]
-        currentColor = currentColor + 1
+DEFAULT_COLORS = [
+    (240, 236, 187),
+    (181, 175, 105),
+    (115, 130, 92),
+    (95, 76, 53),
+]
+
+
+def replace_colors(season_file: str, season: str) -> Image:
+    """
+    Load original sprite and tint it for the season
+    """
+    replacement_colors = SEASON_PREFIXES[season]
+    img = Image.open(season_file).convert('RGB')
+    data = numpy.array(img)
+    current_color = 0
+
+    for color in DEFAULT_COLORS:
+        # TODO: describe
+        data[(data == color).all(axis=-1)] = replacement_colors[current_color]
+        current_color = current_color + 1
```

Comment:
This block is a little hard to explain, but here goes. 
We're iterating through each default (spring) color. We've used numpy to turn the image into an array (var data). As we iterate through each spring color we're also iterating through the replacement colors array. We're basically mass replacing the default colors, starting with the lightest and moving to the darkest.
Afterwards we use numpy to reconstruct the image. This method (replace_colors) is run for each alternative season (summer, fall and winter).